src/Rings/MPolyMap/MPolyQuo.jl

```diff
@@ -93,7 +94,7 @@ function hom(R::MPolyQuoRing, S::NCRing, images::Vector; check::Bool = true)
   @req n == length(images) "Number of images must be $n"
   # Now coerce into S or throw an error if not possible
   imgs = _coerce(S, images)
-  if check
+  @check begin
     _check_imgs_quo(R, S, imgs)
     _check_homo(S, imgs)
```

Comment:
```suggestion
    _check_homo(S, imgs)
    true
```
But actually, I would suggest a different approach: drop these `true` again (readers may wonder why they are they, plus they are very easy to forget, as this example shows), and instead improve the `check` macro to better deal with these `begin` blocks.

Two options come to mind:
1. we could simply accept the return value `nothing` just like we accept `true`; the easiest way would be to replace `$(esc(cond)) || error($(esc(msg)))` by `$(esc(cond)) != false || error($(esc(msg)))`. This way, only something that really returns `false` constitutes a failed check.
2. we could literally check for `begin` expressions and treat those differently than other expressions... not sure that's a good idea, though...